File: TP2/C Ejercicios/Ejercicio 1/EJERCICIO 1/src/main.cpp
#include <Arduino.h>

// Definir los pines para el botón y el LED
const int botonPin = 2; // Pin del botón conectado a D2
const int ledPin = 13;  // Pin del LED conectado a D13

// Variable para almacenar el estado del botón
int estadoBoton;

void setup() {
  // Configurar el pin del botón como entrada
  pinMode(botonPin, INPUT);

  // Configurar el pin del LED como salida
  pinMode(ledPin, OUTPUT);
}

void loop() {
  // Leer el estado del botón
  estadoBoton = digitalRead(botonPin);

  // Si el botón está presionado, encender el LED
  if (estadoBoton == HIGH) {
    digitalWrite(ledPin, HIGH);
  } else {
    digitalWrite(ledPin, LOW);
  }
}

File: TP2/C Ejercicios/Ejercicio 1/EJERCICIO 1/src/main.py
from machine import Pin
import time

# Definir los pines para el botón y el LED
boton_pin = Pin(2, Pin.IN)  # Pin del botón conectado a D2
led_pin = Pin(13, Pin.OUT)   # Pin del LED conectado a D13

while True:
    # Leer el estado del botón
    estado_boton = boton_pin.value()

    # Si el botón está presionado, encender el LED
    if estado_boton == 1:
        led_pin.on()
    else:
        led_pin.off()

    # Esperar un breve período de tiempo antes de volver a leer el botón
    time.sleep(0.1)
